RATapi/inputs.py

```diff
@@ -387,6 +384,35 @@ def make_problem(project: RATapi.Project) -> ProblemDefinition:
     return problem
 
 
+def get_layer_details(project: RATapi.Project) -> list[int]:
+    """Get parameter indices for all layers defined in the project."""
+    hydrate_id = {"bulk in": 1, "bulk out": 2}
+    layer_details = []
+
+    # Get the thickness, SLD, roughness fields from the appropriate model
+    if project.absorption:
+        layer_fields = list(RATapi.models.AbsorptionLayer.model_fields.keys())[1:-2]
+    else:
+        layer_fields = list(RATapi.models.Layer.model_fields.keys())[1:-2]
+
+    for layer in project.layers:
+        for field in layer_fields:
+            if getattr(layer, field) == "":
+                raise ValueError(
+                    f"In the input project, the {field} field of layer {layer.name} does not have a value "
```

Comment:
`f'In the input project, the "{field}" field of layer "{layer.name}" does not have a value '`